### PR TITLE
feat(config): add support for HELTUN HE-HLS01, HE-HT01, HE-RS01

### PR DESCRIPTION
**Pull request by popescunsergiu**
feat(config): Add support for HELTUN devices HE-HLS01 HE-HT01 HE-RS01

### CODE REVIEW

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+			"valueSize": 4,
+			"minValue": 0,
+			"maxValue": 0,
+			"defaultValue": 0,
+			"readOnly": true,
+			"allowManualEntry": false
+		},
+		"7": {
+			"label": "Relay output mode",
+			"description": "NO or NC",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]:
```suggestion
			"label": "Relay Output Mode",
```

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+				},
+				{
+					"label": "Normally Closed",
+					"value": 1
+				}
+			]
+		},
+		"10": {
+			"label": "Floor Sensor Resistance",
+			"description": "External Floor NTC temperature sensor resistance",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]:
```suggestion
			"label": "Floor Temperature Sensor Resistance",
```
Unnecessary description

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+		{
+			"productType": "0x0004",
+			"productId": "0x000a"
+		}
+	],
+	"firmwareVersion": {
+		"min": "0.0",
+		"max": "255.255"
+	},
+	"supportsZWavePlus": true,
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]:
```suggestion
```
That's not used anymore.

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+				},
+				{
+					"label": "Enabled",
+					"value": 1
+				}
+			]
+		},
+		"155": {
+			"label": "Over-Load Protection & Notification",
+			"description": "The value of this parameter can be set from 100 to 4000 in watts. Use the value 0 if there is a need to disable this function.",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]:
```suggestion
			"description": "Allowable Range 100-4000.",
```
and add `Disable` as an option.

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 5]
```diff
@@ -0,0 +1,455 @@
...
+			"unit": "Watts",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]:
```suggestion
			"unit": "W",
```
We try to standardize units: https://zwave-js.github.io/node-zwave-js/#/config-files/style-guide?id=units

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 6]
```diff
@@ -0,0 +1,455 @@
...
+			"label": "Hardware and Software Versions",
+			"valueSize": 4,
+			"minValue": 0,
+			"maxValue": 0,
+			"defaultValue": 0,
+			"readOnly": true,
+			"allowManualEntry": false
+		},
+		"4": {
+			"label": "Energy Consumption, kW",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]:
```suggestion
			"label": "Energy Consumption",
```
Units have a separate field, they don't go into the label or description unless necessary.

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 7]
```diff
@@ -0,0 +1,455 @@
...
+			"unit": "kW",
+			"valueSize": 4,
+			"minValue": 0,
+			"maxValue": 0,
+			"defaultValue": 0,
+			"readOnly": true,
+			"allowManualEntry": false
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]:
```suggestion
			"readOnly": true
```
The other properties shouldn't be necessary for readonly params.

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 8]
```diff
@@ -0,0 +1,455 @@
...
+					"value": 1
+				}
+			]
+		},
+		"17": {
+			"label": "Temperature Sensor Calibration",
+			"valueSize": 1,
+			"minValue": -100,
+			"maxValue": 100,
+			"defaultValue": 0
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]: Missing unit

**Review by AlCalzone — CHANGES_REQUESTED** [review]
Please read the [style guide](https://zwave-js.github.io/node-zwave-js/#/config-files/style-guide) thoroughly and apply it to these files.

I've highlighted **some** examples of what's wrong, but there are more.

File: packages/config/config/devices/0x0344/he-hls01.json (hunk 8 above)
Comment (popescunsergiu) [reply packages/config/config/devices/0x0344/he-hls01.json]: Fixed, Thank you

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 9]
```diff
@@ -0,0 +1,449 @@
...
+			"label": "Floor Temperature Sensor Resistance",
+			"unit": "kΩ",
+			"valueSize": 1,
+			"minValue": 1,
+			"maxValue": 100,
+			"defaultValue": 10
+		},
+		"16": {
+			"label": "Temperature Scale",
+			"description": "Allowable range: 1 Celsius - 2 Fahrenheit",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]:
```suggestion
```

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 10]
```diff
@@ -0,0 +1,449 @@
...
+			"options": [
+				{
+					"label": "Disabled",
+					"value": 0
+				},
+				{
+					"label": "Enabled",
+					"value": 1
+				}
+			]
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]:
```suggestion
			"options": [
				{
					"label": "Disable",
					"value": 0
				},
				{
					"label": "Enable",
					"value": 1
				}
			]
```

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 11]
```diff
@@ -0,0 +1,449 @@
...
+		"24": {
+			"label": "Auto Off Timeout",
+			"unit": "minutes",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 120,
+			"defaultValue": 5
+		},
+		"25": {
+			"label": "Auto On Reconnect Timeout",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json:225]: What does this do? Maybe here a **short** description would be helpful.

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 12]
```diff
@@ -0,0 +1,449 @@
...
+		"41": {
+			"label": "External Input: Hold Control Mode",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 3,
+			"defaultValue": 2,
+			"allowManualEntry": false,
+			"options": [
+				{
+					"label": "Hold function is disabled",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]:
```suggestion
					"label": "Disable",
```

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 13]
```diff
@@ -0,0 +1,449 @@
...
+		"51": {
+			"label": "External Input: Click Control Mode",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 5,
+			"defaultValue": 1,
+			"allowManualEntry": false,
+			"options": [
+				{
+					"label": "Click function disabled",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]:
```suggestion
					"label": "Disable",
```

File: packages/config/config/devices/0x0344/he-hls01.json (hunk 10 above)
Comment (AlCalzone) [reply packages/config/config/devices/0x0344/he-hls01.json]: Please change the other instances aswell.

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 14]
```diff
@@ -0,0 +1,449 @@
...
+		"71": {
+			"label": "Relay Timer Mode Duration",
+			"unit": "seconds",
+			"valueSize": 3,
+			"minValue": 0,
+			"maxValue": 43200,
+			"defaultValue": 0
+		},
+		"141": {
+			"label": "Energy Consumption Meter Consecutive Report Interval",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]:
```suggestion
			"label": "Energy Consumption Report Interval",
```

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 15]
```diff
@@ -0,0 +1,449 @@
...
+		"142": {
+			"label": "Energy Consumption Report Threshold",
+			"unit": "W",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 50,
+			"defaultValue": 25
+		},
+		"143": {
+			"label": "Temperature Sensor Consecutive Report Interval",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]:
```suggestion
			"label": "Temperature Report Interval",
```

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 16]
```diff
@@ -0,0 +1,449 @@
...
+			"unit": "minutes",
+			"valueSize": 1,
+			"minValue": 1,
+			"maxValue": 120,
+			"defaultValue": 10
+		},
+		"144": {
+			"label": "Temperature Sensor Report Threshold",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]:
```suggestion
			"label": "Temperature Report Threshold",
```

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 17]
```diff
@@ -0,0 +1,449 @@
...
+				},
+				{
+					"label": "Enabled",
+					"value": 1
+				}
+			]
+		},
+		"152": {
+			"label": "Temperature Rise Notification",
+			"description": ": Value of this Parameter should be lower than value of Parameter 153 (Overheat Protection & Notification).",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]:
```suggestion
			"description": "The value should be lower than the value of parameter 153 (Overheat Protection & Notification).",
```

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 18]
```diff
@@ -0,0 +1,449 @@
...
+				},
+				{
+					"label": "Enabled",
+					"value": 1
+				}
+			]
+		},
+		"155": {
+			"label": "Over-Load Protection & Notification",
+			"description": "Allowable Range: 100-4000. 0 to disable function.",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]:
```suggestion
			"description": "Allowable Range: 100-4000",
```
Please add a single `option` for `"Disable"`.

Comment (AlCalzone) [reply packages/config/config/devices/0x0344/he-hls01.json]: Same for the next parameters

File: packages/config/config/devices/0x0344/he-ht01.json [hunk 19]
```diff
@@ -0,0 +1,795 @@
...
+		"5": {
+			"label": "Display Brightness Control",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 10,
+			"defaultValue": 0
+		},
+		"6": {
+			"label": "Touch Buttons Sensitivity",
+			"description": "Allowable range: 1 (low sensitivity) to 10 (high sensitivity).",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-ht01.json]:
```suggestion
			"description": "1 = low sensitivity; 10 = high sensitivity",
```

File: packages/config/config/devices/0x0344/he-ht01.json [hunk 20]
```diff
@@ -0,0 +1,795 @@
...
+				},
+				{
+					"label": "MAN",
+					"value": 6
+				}
+			]
+		},
+		"10": {
+			"label": "Floor Sensor Resistance",
+			"unit": "kohm",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-ht01.json]: You used the Ohm symbol above, which I find preferable

File: packages/config/config/devices/0x0344/he-ht01.json [hunk 21]
```diff
@@ -0,0 +1,795 @@
...
+				},
+				{
+					"label": "Time regulator and Floor sensor",
+					"value": 7
+				}
+			]
+		},
+		"12": {
+			"label": "Air Temp Min",
+			"description": "Allowable range: 1 - 36, x10",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-ht01.json]:
```suggestion
```
This is unnecessary with the unit and min/max value definitions. Please also change the next parameters.

File: packages/config/config/devices/0x0344/he-ht01.json [hunk 22]
```diff
@@ -0,0 +1,795 @@
...
+					"value": 6
+				},
+				{
+					"label": "Time regulator and Floor sensor",
+					"value": 7
+				}
+			]
+		},
+		"12": {
+			"label": "Air Temp Min",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-ht01.json]:
```suggestion
			"label": "Air Temperature: Minimum",
```

File: packages/config/config/devices/0x0344/he-ht01.json [hunk 23]
```diff
@@ -0,0 +1,795 @@
...
+			"description": "Allowable range: 1 - 36, x10",
+			"unit": "0.1 °C",
+			"valueSize": 2,
+			"minValue": 10,
+			"maxValue": 360,
+			"defaultValue": 210
+		},
+		"13": {
+			"label": "Air Temp Max",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-ht01.json]:
```suggestion
			"label": "Air Temperature: Maximum",
```
Please change the following parameters in a similar fashion.

File: packages/config/config/devices/0x0344/he-ht01.json [hunk 24]
```diff
@@ -0,0 +1,795 @@
...
+		"22": {
+			"label": "Time Override",
+			"unit": "HHMM",
+			"valueSize": 2,
+			"minValue": 0,
+			"maxValue": 2359,
+			"defaultValue": 0
+		},
+		"23": {
+			"label": "Time Regulation Reg On Time",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-ht01.json]: Please no abbreviations that require a manual to explain them. What's a "Reg On Time"?

File: packages/config/config/devices/0x0344/he-ht01.json [hunk 25]
```diff
@@ -0,0 +1,795 @@
...
+			"unit": "minutes",
+			"valueSize": 2,
+			"minValue": 10,
+			"maxValue": 240,
+			"defaultValue": 30
+		},
+		"24": {
+			"label": "Time Regulation Reg Off Time",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-ht01.json]: dito

File: packages/config/config/devices/0x0344/he-ht01.json [hunk 26]
```diff
@@ -0,0 +1,795 @@
...
+		},
+		"45": {
+			"label": "Monday Morning Temp",
+			"description": "Allowable range: 1 - 37, x10",
+			"unit": "0.1 °C",
+			"valueSize": 2,
+			"minValue": 10,
+			"maxValue": 370,
+			"defaultValue": 240
+		},
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-ht01.json]: See comments above, please apply them here (and the following parameters) too.

File: packages/config/config/devices/0x0344/he-rs01.json [hunk 27]
```diff
@@ -0,0 +1,862 @@
...
+					"value": 0
+				},
+				{
+					"label": "NC",
+					"value": 1
+				}
+			]
+		},
+		"12": {
+			"label": "Relay 1 Power Load",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-rs01.json]: What do these parameters do?

File: packages/config/config/devices/0x0344/he-rs01.json [hunk 28]
```diff
@@ -0,0 +1,862 @@
...
+			"description": "This function can be used to open/close garage doors, blinds, curtains, etc. or to turn attached devices like door lock or security OFF for a short time. If the parameter value is set to ―0‖ it will operate as a short contact (about 0.5sec).",
+			"unit": "seconds",
+			"valueSize": 3,
+			"minValue": 0,
+			"maxValue": 43200,
+			"defaultValue": 0
+		},
+		"72": {
+			"label": "Relay 2 Timer Mode Duration",
+			"description": "This function can be used to open/close garage doors, blinds, curtains, etc. or to turn attached devices like door lock or security OFF for a short time. If the parameter value is set to ―0‖ it will operate as a short contact (about 0.5sec).",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-rs01.json]: This description is too long. The "short" activation on value 0 should be an `option` with label `"0.5 seconds"` instead.

File: packages/config/config/devices/0x0344/he-rs01.json [hunk 29]
```diff
@@ -0,0 +1,862 @@
...
+				},
+				{
+					"label": "Relay 1 and Relay 2",
+					"value": 12
+				},
+				{
+					"label": "Relay 2 and Relay 3",
+					"value": 23
+				}
+			]
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-rs01.json]: These values don't seem to make sense. Why is the max 51 when you can only select 0, 12 and 23?

**Review by AlCalzone — CHANGES_REQUESTED** [review]
Thanks, this is better, but there are a few more things. Anything I've highlighted in one file also applies to the other files aswell.

File: packages/config/config/devices/0x0344/he-rs01.json (hunk 29 above)
Comment (popescunsergiu) [reply packages/config/config/devices/0x0344/he-rs01.json]: Fixed

File: packages/config/config/devices/0x0344/he-rs01.json (hunk 28 above)
Comment (popescunsergiu) [reply packages/config/config/devices/0x0344/he-rs01.json]: Fixed

File: packages/config/config/devices/0x0344/he-ht01.json (hunk 25 above)
Comment (popescunsergiu) [reply packages/config/config/devices/0x0344/he-ht01.json]: Added description, fixed label

File: packages/config/config/devices/0x0344/he-hls01.json (hunk 11 above)
Comment (popescunsergiu) [reply packages/config/config/devices/0x0344/he-hls01.json:225]: Added description

File: packages/config/config/devices/0x0344/he-ht01.json (hunk 26 above)
Comment (popescunsergiu) [reply packages/config/config/devices/0x0344/he-ht01.json]: removed description

File: packages/config/config/devices/0x0344/he-ht01.json (hunk 20 above)
Comment (popescunsergiu) [reply packages/config/config/devices/0x0344/he-ht01.json]: fixed

File: packages/config/config/devices/0x0344/he-ht01.json (hunk 24 above)
Comment (popescunsergiu) [reply packages/config/config/devices/0x0344/he-ht01.json]: Added description and fixed label

**Comment by steve-hodgkiss** [comment]
Hi - noticed a few entries contain Temperatureerature as a typo.

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 30]
```diff
@@ -0,0 +1,446 @@
...
+			"label": "High Load Timeout Protection: Power Threshold",
+			"unit": "W",
+			"valueSize": 2,
+			"minValue": 100,
+			"maxValue": 3500,
+			"defaultValue": 0
+		},
+		"27": {
+			"label": "High Load Timeout Protection: Time Threshold",
+			"unit": "W",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json]: This should be minutes, right?

File: packages/config/config/devices/0x0344/he-hls01.json [hunk 31]
```diff
@@ -0,0 +1,446 @@
...
+			"label": "High Load Timeout Protection: Power Threshold",
+			"unit": "W",
+			"valueSize": 2,
+			"minValue": 100,
+			"maxValue": 3500,
+			"defaultValue": 0
+		},
+		"27": {
+			"label": "High Load Timeout Protection: Time Threshold",
+			"unit": "minutes",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0344/he-hls01.json:243]: To be sure, this was supposed to be `"minutes"`, right? Your file had `"W"`.
@popescunsergiu

**Review by AlCalzone — DISMISSED** [review]
I've changed a few labels etc. Looks good now.

**Comment by johanschelin** [comment]
Hi @popescunsergiu! Excellent that you are helping with Heltun Devices, as I also run several of them.
I am also running their models FT01, TPS01. TPS02, TPS03, and TPS05, and are not used to create these files, are you able to help me? :)
@AlCalzone - is it appropriate for us to help with these models in the same issue or, should we open a new one so this one could be closed?

**Comment by AlCalzone** [comment]
@johanschelin lets make the remaining ones a new PR - this one is already pretty long.

File: packages/config/config/devices/0x0344/he-hls01.json (hunk 31 above)
Comment (popescunsergiu) [reply packages/config/config/devices/0x0344/he-hls01.json:243]: Yes, sorry for missing this

**Comment by popescunsergiu** [comment]
Hello @AlCalzone , is there anything else I'll have to fix on this PR? Please point anything else that blocks the merge. I plan to work on few other devices but I'd like to finish my 1st PR first. Thx

**Comment by AlCalzone** [comment]
No, with your confirmation, this is now ready to merge.

**Review by AlCalzone — APPROVED** [review]

**Pull request merged**